### ISSUE TITLE
Impressions Discrepancy with IndexExchange When moved to Prebid 1.11.0

### ISSUE THREAD

**Issue by gitsh84** [prebid/Prebid.js#2690]
## Type of issue
We see a correlation with our deployment of Prebid 1.11.0 (previously 0.29) and a sudden rise in discrepancy in number of impressions (and revenues) counted by our DFP and Index Dashboard.

## Description
A few days prior (!) to GDPR we moved all of our sites (not on the same day) from Prebid 0.29 to Prebid 1.11.
Comparing our DFP data with Index data for each site there is a clear correlation between the day of deployment and a sudden rise in discrepancy (around %50 impressions less).

(Note #1 that we are counting both "indexExchange" and "ix" bidders as they switched names in Prebid 1+. As expected we can see that  indexExchange impressions have disappeared and ix started.
In other words what I'm saying is that I don't think the name change is related here.)

(Note #2 the discrepancy started a few days before GDPR took effect so  I don't think it's related either)

Would appreciate anybody who understand the changes done in Prebid 1+ to maybe help out and shed some light here re what change/feature/bug in the new version could produce such a discrepancy.

Multiple other bidders ? Changes with are related to same page ad refresh ? Auction timeout mechanism 
changes ? etc.

Thanks !

**Comment by Deimos01** [comment]
Hello @gitsh84,

I think your issue is closely related to that one : #2648 (new concept of concurrent auctions)

**Comment by jaiminpanchal27** [comment]
@gitsh84 this issue is same as #2648. Can you close this one and add your comment in #2648. ?
Also share your test page or example on jsfiddle.

That will keep the discussion in one place. 
Thanks.

**Comment by dmitriyshashkin** [comment]
@jaiminpanchal27 this is clearly not the same issue

1) Problem is limited to indexExchange while in #2648 it is stated that the problem affects all SSPs
2) #2648 mentions the decrease in the number of impressions as reported by DFP. We do not see any decrease in the number of impressions reported by DFP. While a/b testing 0.34 vs 1.x we saw no decrease in impressions or revenue. The problem is that indexExchange sees 50% fewer impressions than DFP or our own analytical adapter for prebid

**Comment by dmitriyshashkin** [comment]
@gitsh84 mind if ask you, do you refresh ads? Or mb you have single page application? We see the most severe impact on the units we refresh the most.

**Comment by jaiminpanchal27** [comment]
@gitsh84 There was this issue https://github.com/prebid/Prebid.js/issues/2574. I have merged the fix. It will be available in next release.

**Comment by mercuryyy** [comment]
We refresh ads and see the same problem with IX new 1.x adapter.

**Comment by dmitriyshashkin** [comment]
@jaiminpanchal27 please read the description of the bug you've mentioned. It has nothing to do with the discrepancy. When media type was omitted Index was ignoring this unit altogether. It did not bid on this unit, it did not win impressions on this unit and therefore there was no basis for the discrepancy. As I've said in the description of the bug "Index excludes this unit from bid request. "Imp" field in https request to cygnus server is empty"

**Comment by Caspervw** [comment]
@dmitriyshashkin Can confirm, our platform has the same discrepancy. 50% difference, Prebid 1.13, Single Page Application, Refreshing turned on.

@ix-prebid-support Is this on your radar?

**Comment by dmitriyshashkin** [comment]
BTW, we've excluded index from refresh auctions and our discrepancy went back to normal.

**Comment by dmitriyshashkin** [comment]
@ix-prebid-support any progress with this?

**Comment by rmertens** [comment]
we also see this issue with Prebid 1.15 (around 40% discrepancy both in impressions and revenue). We refresh as well. @ix-prebid-support any news on this?

**Comment by ix-prebid-support** [comment]
@rmertens  @dmitriyshashkin 
We have an update to our adapter in #2744 which we believe would address this issue.

**Comment by dmitriyshashkin** [comment]
@ix-prebid-support how exactly this update will address the issue? As far as I can tell here is the summary of the changes:
1) If "id" property is set - use it as a sid, otherwise, construct sid from size. Probably has nothing to do with the problem as "id" is not a mandatory parameter
2) Do not drop the request if the media type is not set explicitly aka "Backward compatible with legacy config". OK, first of all, we did specify media type explicitly and that did not help with the discrepancy. Secondly, omitting media type DID NOT result in a discrepancy. The bid request was dropped completely as if index adapter was completely turned off.
3) You've changed ttl. We render bids as soon as the auction is over, 2-3 seconds after the bid response has been sent. We do not store or cache bid responses. Ttl has nothing to do with our problem.

Please correct me if I missed something. But apparently your PR does not address the issue and we'll have to wait another month for indexExchange to actually do something.

**Comment by ix-prebid-support** [comment]
@dmitriyshashkin 

The creative TTL is being used by the Prebid framework to filter out expired bids and what we observed was that due to the current TTL configuration there were often cases where the bids which were already expired in previous auctions were being reapplied and shortening the TTL resolved the issue

**Comment by dmitriyshashkin** [comment]
@ix-prebid-support sorry for being too skeptical. Reduction of TTL did in fact solved the issue. Ran a test and discrepancy is down to 3-4%. Thank you!

Still, now it seems that there is another issue within prebid's core that allowed bids from previous auctions to participate. Or is it an expected behavior?

**Comment by Deimos01** [comment]
@ix-prebid-support Thanks for your explanation. I didn't know that the TTL parameter is also used by the Prebid framework to filter out expired bids.

I agree with @dmitriyshashkin, If the reduction of the TTL solved the issue, it means there is an other issue because Prebid shouldn't use bids from previous auctions since this fix : #2676 

What do you think @jaiminpanchal27 ?

**Comment by ix-certification** [comment]
@gitsh84 
@mercuryyy 
@Caspervw 
@rmertens 

It would be great if you can let us know if the following update has addressed the refresh issue you observed and we would be able to close this investigation.

**Comment by rmertens** [comment]
The TTL was the issue for us so it's solved!

On Sat, Jul 14, 2018 at 3:26 AM, ix-prebid-support <notifications@github.com

**Comment by ix-prebid-support** [comment]
@gitsh84
@mercuryyy
@Caspervw

Following up on the earlier request, we would really appreciate it if you can confirm that this fix has resolved the issue with impression discrepancy

**Comment by Caspervw** [comment]
@ix-prebid-support Yes, this solved it for us. Thanks
